### PR TITLE
Refactor some code in the grades view add additional error handling

### LINKED ISSUE #154: Refactor some code in the grades view add additional error handling
There was text in the grades view 
"You don\'t seems to have your grade yet"

That should be rewritten, also this is redundant and also there was a problem in the view when there was no grades calculated in the database.

### PR DESCRIPTION
**Pull request by jonespm**
fixes #154

### CODE REVIEW

File: dashboard/views.py [hunk 1]
```diff
@@ -213,7 +213,7 @@ def grade_distribution(request, course_id=0):
     grade_score_sql = "select current_grade,(select current_grade from user where sis_name=" \
                       "%(current_user)s and course_id=%(course_id)s) as current_user_grade from user where course_id=%(course_id)s;"
     df = pd.read_sql(grade_score_sql, conn, params={"current_user": current_user,'course_id': course_id})
-    if df.empty:
+    if df.empty or df['current_grade'].isnull().all():
```
Comment (pushyamig) [inline dashboard/views.py]: Nice catch! all 3 courses have grades come in in my local DB, so I couldn't catch this bug

File: dashboard/static/js/grade_distribution.js [hunk 2]
```diff
@@ -158,20 +158,16 @@ var makeGraph = function () {
...
-                .html('<span>Number of students: <b>' + totalStudents + '</b></span><br>' +
-                    '<span>Average grade: <b>' + gradeAverage + '</b></span><br>' +
-                    '<span>Standard deviation of grades: <b>' + standardGradeDeviation + '</b></span><br>')
+        var gradeDetailsHtml = 
+            '<span>Number of students: <b>' + totalStudents + '</b></span><br>' +
+            '<span>Average grade: <b>' + gradeAverage + '</b></span><br>' +
+            '<span>Standard deviation of grades: <b>' + standardGradeDeviation + '</b></span><br>';
+        if(myscore==null) {
+            gradeDetailsHtml += '<span><b>There are no grades yet for you in this course</b></span><br>';
         }
```
Comment (pushyamig) [inline dashboard/static/js/grade_distribution.js]: Nice

File: dashboard/views.py (hunk 1 above)
Comment (jonespm) [reply dashboard/views.py]: Yeah thanks I noticed a stack trace locally when there were no grades for this user at all. This caught that case.

Comment (pushyamig) [reply dashboard/views.py]: I feel the stack trace would have occurred when there are no grades at all for the course vs no grade for user. The code is filtering out the null's and if user grade is null then it is shown as message of "no grades for you"

Comment (jonespm) [reply dashboard/views.py]: Yeah, that might have been what happened too. Basically if the nulls are all filtered out and df was empty after that process was done. This just catches it a little bit earlier.

**Pull request merged**